packages/node/src/integrations/tracing/redis.ts

```diff
@@ -1,14 +1,85 @@
 import { IORedisInstrumentation } from '@opentelemetry/instrumentation-ioredis';
-import { defineIntegration } from '@sentry/core';
+import {
+  SEMANTIC_ATTRIBUTE_CACHE_HIT,
+  SEMANTIC_ATTRIBUTE_CACHE_ITEM_SIZE,
+  SEMANTIC_ATTRIBUTE_CACHE_KEY,
+  SEMANTIC_ATTRIBUTE_SENTRY_OP,
+  defineIntegration,
+  spanToJSON,
+} from '@sentry/core';
 import { addOpenTelemetryInstrumentation } from '@sentry/opentelemetry';
 import type { IntegrationFn } from '@sentry/types';
 
-const _redisIntegration = (() => {
+function keyHasPrefix(key: string, prefixes: string[]): boolean {
+  return prefixes.some(prefix => key.startsWith(prefix));
+}
+
+/** Currently, caching only supports 'get' and 'set' commands. More commands will be added (setex, mget, del, expire) */
+function shouldConsiderForCache(
+  redisCommand: string,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  key: string | number | any[] | Buffer,
+  prefixes: string[],
+): boolean {
+  return (redisCommand === 'get' || redisCommand === 'set') && typeof key === 'string' && keyHasPrefix(key, prefixes);
+}
+
+function calculateCacheItemSize(response: unknown): number {
+  try {
+    if (Buffer.isBuffer(response)) return response.byteLength;
+    else if (typeof response === 'string') return response.length;
+    else if (typeof response === 'number') return response.toString().length;
+    else if (response === null || response === undefined) return 0;
+    return JSON.stringify(response).length;
+  } catch (e) {
+    return 0;
+  }
+}
+
+interface RedisOptions {
+  cachePrefixes?: string[];
+}
+
+const _redisIntegration = ((options?: RedisOptions) => {
   return {
     name: 'Redis',
     setupOnce() {
       addOpenTelemetryInstrumentation([
-        new IORedisInstrumentation({}),
+        new IORedisInstrumentation({
+          responseHook: (span, redisCommand, cmdArgs, response) => {
+            const key = cmdArgs[0];
+
+            if (!options?.cachePrefixes || !shouldConsiderForCache(redisCommand, key, options.cachePrefixes)) {
+              // not relevant for cache
+              return;
+            }
+
+            const networkPeerAddress = spanToJSON(span).data?.['net.peer.name'];
```

Comment:
out of interest, are we using something non-standard here or is redis emitting something non-standard?

Reply:
redis seems to be using the old standard as there was a change to those params: https://github.com/open-telemetry/opentelemetry-specification/issues/3199

We are using those: https://opentelemetry.io/docs/specs/semconv/attributes-registry/network/

Reply:
ok, got it, makes sense - maybe we can leave a comment like this in the code so we can potentially revisit this when we update the redis instrumentation at some point :)